os/arch/arm/src/amebasmart/amebasmart_spi.c

```diff
@@ -1002,8 +1135,66 @@ static void amebasmart_spi_exchange_nodma(FAR struct spi_dev_s *dev,
 			}
 		}
 	}
+#else
+/*************************************************************************
+ * Name: spi_exchange (with DMA)
+ *
+ * Description:
+ *   Exchange a block of data on SPI using DMA
+ *
+ * Input Parameters:
+ *   dev      - Device-specific state data
+ *   txbuffer - A pointer to the buffer of data to be sent
+ *   rxbuffer - A pointer to a buffer in which to receive data
+ *   nwords   - the length of data to be exchanged in units of words.
+ *              The wordsize is determined by the number of bits-per-word
+ *              selected for the SPI interface.  If nbits <= 8, the data is
+ *              packed into uint8_t's; if nbits >8, the data is packed into uint16_t's
+ *
+ * Returned Value:
+ *   None
+ *
+ ************************************************************************************/
+	uint8_t *rxbuff = NULL;
+	rxbuff = (uint8_t *)rtw_zmalloc(65535);
```

Comment:
How about #define CONFIG_SPI_DMA_MAX_BUFFER_SIZE 65535, without a predecessor definition, it is hard to understand what it means here.